src/Extension/ExtensionInterface.php

```diff
@@ -39,4 +41,9 @@ public function getAllowedProperties(): array;
      * @return array<class-string, array<string>>
      */
     public function getAllowedMethods(): array;
+
+    /**
+     * @return array<string, class-string<RuntimeExtensionInterface>>
```

Comment:
It might worth to explain the shape of the array. Keys need to be the service ids, it's not obvious.

Reply:
We might only provide the Id and I can retrieve that Id from the container, that would be better?

Reply:
All the methods are returning string[] or string[][]. So It might be better to have a similar signature indeed.

Reply:
Explained